### PR TITLE
ENCD-5955-analysis-titles-in-file-details-table-headers

### PR DESCRIPTION
**Pull request by forresttanaka**

### CODE REVIEW

File: src/encoded/static/scss/encoded/modules/_panels.scss [hunk 1]
```diff
@@ -767,6 +769,25 @@ $file-gallery-controls-height: 34px; // Height of controls inside file-gallery-c
     color: #606060;
 }
 
+.title-main-content {
```
Comment (zoldello) [inline src/encoded/static/scss/encoded/modules/_panels.scss:772]: I am surprised you did not use '&' to capture the "title-main-content" part of the name.

File: src/encoded/static/components/filegallery.js [hunk 2]
```diff
@@ -3748,6 +3744,226 @@ FileGalleryRendererComponent.contextTypes = {
 const FileGalleryRenderer = auditDecor(FileGalleryRendererComponent);
 
 
+/**
+ * Maps each Series-object subtype to a function that generates a subtitle in the file details
+ * table section title bars.
+ */
```
Comment (zoldello) [inline src/encoded/static/components/filegallery.js:3831]: can this be on top of the file?

File: src/encoded/static/components/filegallery.js [hunk 3]
```diff
@@ -503,7 +503,6 @@ export class FileTable extends React.Component {
                                             && filterDownloadableFilesByStatus(context, files[key]).length > 0
                                         }
                                         inclusionOn={options.inclusionOn}
-                                        files={files[key]}
```
Comment (zoldello) [inline src/encoded/static/components/filegallery.js]: Not sure why you removed this. I do not see functionality-replacement (for non-series objects).

File: src/encoded/static/components/filegallery.js [hunk 4]
```diff
@@ -3748,6 +3744,226 @@ FileGalleryRendererComponent.contextTypes = {
...
+}) => {
+    // Based on the @type of the dataset, get a composer function appropriate for it to generate
+    // a subtitle, passing it the related datasets if it exists. Any dataset objects without a
+    // composer function get rendered normally.
+    let subtitle;
+    const type = context['@type'][0];
+    if (context.related_datasets && context.related_datasets.length > 0 && analysis) {
+        const relatedDataset = context.related_datasets.find((dataset) => analysis.datasets.includes(dataset['@id']));
+        if (relatedDataset) {
+            const composer = subtitleComposers[type];
```
Comment (zoldello) [inline src/encoded/static/components/filegallery.js:3988]: Where is the logic for reading series without a composer?

**Review by zoldello — DISMISSED** [review]
I have some questions ...

File: src/encoded/static/components/filegallery.js (hunk 4 above)
Comment (forresttanaka) [reply src/encoded/static/components/filegallery.js:3988]: Without a composer, you get an empty `subtitle`, and the title gets renders like it does in production today. That applies to all non-Series objects too.

Comment (zoldello) [reply src/encoded/static/components/filegallery.js:3988]: thanks.

**Comment by forresttanaka** [comment]
Updating with Idan’s latest spec.

**Pull request closed without merging**